<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Grammar</class>
 <widget class="QWidget" name="Grammar">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>469</width>
    <height>375</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QPushButton" name="loadGrammar">
   <property name="geometry">
    <rect>
     <x>300</x>
     <y>50</y>
     <width>156</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Задать грамматику из файла</string>
   </property>
  </widget>
  <widget class="QPushButton" name="updateAlpha">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>20</y>
     <width>105</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Обновить алфавит</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>0</y>
     <width>121</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Алфавит</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineTerm">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>90</y>
     <width>75</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Список правил</string>
   </property>
  </widget>
  <widget class="QListWidget" name="rulesList">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>125</y>
     <width>449</width>
     <height>211</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="saveGrammar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>340</y>
     <width>125</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Сохранить грамматику</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="fromNterm">
   <property name="geometry">
    <rect>
     <x>19</x>
     <y>50</y>
     <width>25</width>
     <height>23</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>25</width>
     <height>25</height>
    </size>
   </property>
   <property name="maxLength">
    <number>1</number>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="arrow">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>50</y>
     <width>16</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;➜&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="textFormat">
    <enum>Qt::AutoText</enum>
   </property>
   <property name="scaledContents">
    <bool>false</bool>
   </property>
   <property name="wordWrap">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="toChain">
   <property name="geometry">
    <rect>
     <x>69</x>
     <y>51</y>
     <width>101</width>
     <height>20</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>100</width>
     <height>0</height>
    </size>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="addRule">
   <property name="geometry">
    <rect>
     <x>190</x>
     <y>50</y>
     <width>103</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Добавить правило</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>0</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Нетерминальные</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineNonterm">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>20</y>
     <width>81</width>
     <height>20</height>
    </rect>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>290</x>
     <y>0</y>
     <width>120</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Символ пустой цепочки</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="emptySym">
   <property name="geometry">
    <rect>
     <x>300</x>
     <y>20</y>
     <width>25</width>
     <height>20</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>25</width>
     <height>25</height>
    </size>
   </property>
   <property name="maxLength">
    <number>1</number>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>190</x>
     <y>0</y>
     <width>140</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Начальный символ</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="startNterm">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>20</y>
     <width>25</width>
     <height>20</height>
    </rect>
   </property>
   <property name="maximumSize">
    <size>
     <width>25</width>
     <height>25</height>
    </size>
   </property>
   <property name="maxLength">
    <number>1</number>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
